### PR TITLE
Example for creating quantum circuit from unitary matrix

### PR DESCRIPTION
**Pull request by smturro2**
### Summary
The pull request fixes issue https://github.com/https://github.com/Qiskit/qiskit-terra/issues/7206. I have added the suggested code chunk to the class description.

### CODE REVIEW

File: qiskit/extensions/unitary.py [hunk 1]
```diff
@@ -35,7 +35,22 @@
 
 
 class UnitaryGate(Gate):
-    """Class for representing unitary gates"""
+    """Class for representing unitary gates
+
+    Example:
+
+        Create a quantum circuit from a unitary matrix
```
[the remark this answers is not in the document]
Comment (smturro2) [reply qiskit/extensions/unitary.py]: Thanks for reviewing this so fast @Cryoris! I moved the original example to the unitary method and then followed your suggestion for the example for the unitary class.

File: qiskit/extensions/unitary.py [hunk 2]
```diff
@@ -215,7 +236,22 @@ def validate_parameter(self, parameter):
 
 
 def unitary(self, obj, qubits, label=None):
-    """Apply unitary gate to q."""
+    """Apply unitary gate to q.
+
+    Example:
+
+        Create a quantum circuit from a unitary matrix
```
Comment (Cryoris) [inline qiskit/extensions/unitary.py:243]:
```suggestion
    """Apply unitary gate specified by ``obj`` to ``qubits``.

    Example:

        Apply a gate specified by a unitary matrix to a quantum circuit
```

File: qiskit/extensions/unitary.py [hunk 3]
```diff
@@ -35,7 +35,28 @@
...
 
 class UnitaryGate(Gate):
-    """Class for representing unitary gates"""
+    """Class for representing unitary gates
+
+    Example:
+
+        Creating a unitary gate from a unitary matrix then adding this
+        gate to a quantum circuit. The matrix can also be added directly
+        to the quantum circuit see the example in the unitary function
```
Comment (Cryoris) [inline qiskit/extensions/unitary.py:44]:
```suggestion
    """Class quantum gates specified by a unitary matrix.

    Example:

        We can create a unitary gate from a unitary matrix then add it
        to a quantum circuit. The matrix can also be directly applied
        to the quantum circuit, see :meth:`~qiskit.QuantumCircuit.unitary`.
```

**Review by Cryoris — APPROVED** [review]
Thanks for the improvement @smturro2!

**Pull request merged**